### PR TITLE
Sort supported versions up within same patch group

### LINKED ISSUE #1034: In the update version dialog show firstly the supported version and then the others
**What would you like to be added**:
When I tried to update a k8s version of a shoot cluster, I see this dialog 
image

It will be nice the supported version to be always on top and after that the deprecated versions to be shown.
For the preview versions, I have no opinion whether they should be before or after the supported version.

**Why is this needed**:
To achieve better user experience, as the supported version will be always the first choice.

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
Sort supported versions up within same patch group, use colors to highlight supported / deprecated versions

Screen Shot 2022-06-28 at 15 55 06

**Which issue(s) this PR fixes**:

Fixes #1034

**Special notes for your reviewer**:

**Release note**:

```feature user
Sort supported versions up within same patch group, use colors to highlight supported / deprecated versions
```

### CODE REVIEW

**Comment by grolu** [comment]
Instead of using the success color for the supported version, we could also make it bold.

File: frontend/src/components/ShootVersion/ShootVersionUpdate.vue [hunk 1]
```diff
@@ -93,14 +101,21 @@ export default {
...
           } else if (b.header) {
             return 1
-          } else {
-            if (semver.eq(a.version, b.version)) {
-              return 0
-            } else if (semver.gt(a.version, b.version)) {
-              return 1
-            } else {
+          }
+          if (semver.diff(a.version, b.version) === 'patch') {
```
Comment (holgerkoser) [inline frontend/src/components/ShootVersion/ShootVersionUpdate.vue:105]: The versions are already grouped by update type `['patch', 'minor', 'major']`. I would propose to sort the versions for each group and afterwards add the header to the group. In this case all the entries would not need to have the type property I think.

Comment (holgerkoser) [reply frontend/src/components/ShootVersion/ShootVersionUpdate.vue:105]: How do we handle `pre` release versions?

**Review by petersutter — APPROVED** [review]
/lgtm

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Pull request merged**